scripts/pull-gee-repository.sh

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env bash
+
+#
+# Copyright 2017 Open GEE Contributors
+#
+# Licensed under the Apache license, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+#===============================================================================
+#
+#          FILE:  pull-gee-repository.sh
+# 
+#         USAGE:  ./pull-gee-repository.sh 
+# 
+#   DESCRIPTION:  Acts as a frontend to allow for easier cloning of codebase 
+# 
+#  REQUIREMENTS:  Must not be root 
+#       VERSION:  1.0
+#===============================================================================
+
+
+#===============================================================================
+#       Constants  
+#===============================================================================
+THIS=$(basename "$0")
+USER_NAME=""
+PASSWORD=""
+DIR_NAME="earthenterprise"
+IS_SSH=1
+IS_HTTPS=2
+IS_MAIN=3
+IS_FORK=4
+SSH_DIR=".ssh"
+SSH_KEY_PRIV="id_rsa"
+SSH_KEY_PUB="$SSH_KEY_PRIV.pub"
+CLONE_REPO=$IS_MAIN
+CLONE_TYPE=""
+
+#===  FUNCTION  ================================================================
+#          NAME:  is_cloned
+#   DESCRIPTION:  Determines if already cloned
+#    PARAMETERS:  none
+#       RETURNS:  1, if cloned; 0, if not cloned
+#===============================================================================
+function is_cloned ()
+{
+    [ -d "$DIR_NAME" ];
+}
+
+
+#===  FUNCTION  ================================================================
+#          NAME:  get_type 
+#   DESCRIPTION:  returns the repo to clone, main or fork 
+#    PARAMETERS:  n/a
+#       RETURNS:  IS_FORK, if fork; IS_MAIN, if main
+#===============================================================================
+function get_type ()
+{
+    printf "Do you want to clone the (m)ain repo or a (f)ork (default is main)? "
```

Comment:
This reads to me like "main" would be a valid input here... but the check a couple of lines down will *only* match if someone *exactly* enters an "m".  I'm not terribly proficient with bash, is there a smoother way to do this?